File: Python/01_Strings_Arrays/IsOneEdit.py
def is_one_edit(s1, s2):
    len1 = len(s1)
    len2 = len(s2)
    abs_len_diff = abs(len1-len2)
    if (abs_len_diff > 1):
        return False

    if (abs_len_diff == 0):
        diffFound = False
        for i in range(0, len1):
            if s1[i] != s2[i]:
                if diffFound:
                    return False
                diffFound = True
        return True
    else:
        skipped = False
        shorter_len = len1 if (len1 < len2) else len2
        j = 0
        i = 0
        while i < shorter_len:
            if (s1[i] != s2[j]):
                if skipped:
                    return False
                skipped = True
                if len1 < len2:
                    j += 1
                else:
                    i += 1
            else:
                j+=1
                i+=1
        return True

File: C++/Practice_Pad/source/CH1_Arrays_Strings/is_one_edit.cpp
#include "pch.h"

#include "is_one_edit.h"

bool is_one_edit(string s1, string s2) {

    int len1 = s1.length();
    int len2 = s2.length();
    int abs_len_diff = abs(len1 - len2);

    if (abs_len_diff > 1) {
        
        return false;
    }

    if (abs_len_diff == 0) {
        bool diffFound = false;
        for (int i = 0; i < len1; i++) {
            if (s1.at(i) != s2.at(i)) {
                if (diffFound) {
                    
                    return false;
                }
                diffFound = true;
            }
        }
        
        return true;
    }
    else {
        bool skipped = false;
        int j = 0;
        int shorter_len = (len1 > len2) ? len2 : len1;
        for (int i = 0; i < shorter_len; i) {

            if (s1.at(i) != s2.at(j)) {
                if (skipped) {
                    
                    return false;
                }

                skipped = true;
                if (len1 > len2) {
                    i++;
                }
                else {
                    j++;
                }
            }
            else {
                i++;
                j++;
            }
        }
        
        return true;
    }
}